### ISSUE TITLE
macos failure to run

### ISSUE THREAD

**Issue by FlorinAndrei** [Nuitka/Nuitka#789]
```
$ python -m nuitka --version
0.6.8.4
Python: 3.7.7 (default, Mar 23 2020, 17:31:31) 
Executable: /Users/florin/opt/anaconda3/bin/python
OS: Darwin
Arch: x86_64
```

Installation:

```
conda install -c conda-forge nuitka
```

Hello World fails. This is the build error:

```
[...]
clang -o static_src/MetaPathBasedLoader.o -c -fvisibility=hidden -std=c11 -fwrapv -w -fvisibility=hidden -fvisibility-inlines-hidden -O3 -pipe -D_XOPEN_SOURCE -D_NUITKA_STANDALONE -D_NUITKA_SYSFLAG_BYTES_WARNING=0 -D_NUITKA_SYSFLAG_NO_SITE=0 -D_NUITKA_SYSFLAG_VERBOSE=0 -D_NUITKA_SYSFLAG_UTF8=0 -D_NUITKA_SYSFLAG_OPTIMIZE=0 -D_NUITKA_SYSFLAG_NO_RANDOMIZATION=0 -D__NUITKA_NO_ASSERT__ -D_NUITKA_FROZEN=165 -D_NUITKA_EXE -I/Users/florin/opt/anaconda3/include/python3.7m -I. -I/Users/florin/opt/anaconda3/lib/python3.7/site-packages/nuitka/build/include -I/Users/florin/opt/anaconda3/lib/python3.7/site-packages/nuitka/build/static_src static_src/MetaPathBasedLoader.c
scons: building `/Users/florin/HelloWorld/hello.dist/hello.exe' because it doesn't exist
clang -o /Users/florin/HelloWorld/hello.dist/hello.exe -Wno-deprecated-declarations -headerpad_max_install_names @"./@sources.tmp" -L/Users/florin/opt/anaconda3/lib -lm -lpython3.7m
scons: done building targets.
Error, scons failed to create the expected file 'hello.dist/hello'.
```

And this is the runtime error:

```
$ ./hello.dist/hello.exe 
dyld: Library not loaded: @rpath/libpython3.7m.dylib
  Referenced from: /Users/florin/HelloWorld/./hello.dist/hello.exe
  Reason: image not found
Abort trap: 6
```

**Comment by FlorinAndrei** [comment]
Same Nuitka version, same test scenario, work fine on Windows 10.

**Comment by kayhayen** [comment]
The `.exe` suffix is wrong. Nuitka aborted and didn't apply necessary changes. Using the .dist folder of a stopped compilation makes no sense.

It's supposed to only be used on Windows though.

**Comment by kayhayen** [comment]
Can it be that scons default to ".exe" there, I am confused. Can you debug, where it's coming from and/or also provide your full command line for hello world? Esp. where with `--show-scons ` it's calling Scons.

**Comment by FlorinAndrei** [comment]
Messages from the build command:

https://gist.github.com/FlorinAndrei/f50d25bfd92ebeee3c7aa2fa702e6890

**Comment by kayhayen** [comment]
Thanks, that's very good. So Nuitka main is doing the right thing and passing this: `result_exe=/Users/florin/HelloWorld/hello.dist/hello` on the command line.

The only use of that towards scons is really this: `target = env.Program(result_exe, source_files + source_targets)` in `SingleExe.scons`

Then it says though: 

> Scons: Launching target: ['/Users/florin/HelloWorld/hello.dist/hello.exe']

And the only way, this could really happen is `PROGSUFFIX` which in the inline copy, a few platforms set (in the same way as Nuitka does).

What could be going on this that this code, passes an environment variable that overwrites `PROGSUFFIX`.

```
        # We want the outside environment to be passed through.
        ENV=os.environ,
```

Is is possible that you have this variable `PROGSUFFIX` set?

**Comment by kayhayen** [comment]
My attempts with setting it on Linux, had no influence, not even if I added it to the environment, but maybe it's different for clang, but I didn't find that to make a difference.

Since you use the inline copy of scons, which should be good, I am very confused now, how this `.exe` suffix is invented and added. An output of `env` would be interesting.

**Comment by FlorinAndrei** [comment]
Here's my shell env (I will delete it after a while):

https://gist.github.com/FlorinAndrei/3e3f06d0fa1625eb180e7776a92ce2d8

**Comment by levesquejf** [comment]
I have the same `dyld: Library not loaded: @executable_path/Python` error.

```
nuitka3 --version
0.6.11.5
Python: 3.8.6 (v3.8.6:db455296be, Sep 23 2020, 13:31:39)
Executable: /Users/jfl/.local/share/virtualenvs/nuitka-DfCoynEi/bin/python
OS: Darwin
Arch: x86_64
```

Running macOS 11.2.

main.py is:
```
print("hello world")
```

Result of the compilation with `nuitka3 main.py --follow-imports --standalone --remove-output --clang --show-scons` is at https://gist.github.com/levesquejf/1271d6b4985dea60b84abae39af777cb.

When executing:

```
$ ./main.dist/main
dyld: Library not loaded: @executable_path/Python
  Referenced from: /Users/jfl/tmp/nuitka/./main.dist/main
  Reason: no suitable image found.  Did find:
	/Users/jfl/tmp/nuitka/./main.dist/Python: code signature invalid for '/Users/jfl/tmp/nuitka/./main.dist/Python'

	/Users/jfl/tmp/nuitka/main.dist/Python: code signature invalid for '/Users/jfl/tmp/nuitka/main.dist/Python'

	/Users/jfl/tmp/nuitka/main.dist/Python: stat() failed with errno=1
[1]    67394 abort      ./main.dist/main
```

Let me know if I can help debugging.

**Comment by kayhayen** [comment]
@levesquejf what would be best would be access to a VM on the net where I can reproduce this, let me know if you can provide it.

**Comment by levesquejf** [comment]
@kayhayen I can provide you with a VM running macOS. Let me know when you are ready and I will send you the credentials by email.

**Comment by kayhayen** [comment]
@levesquejf thanks for your support, I did that.

**Comment by kayhayen** [comment]
So, on factory, I sucessfully used 4 Pythons in that VM to create a simple standalone program:

a) The Python2 from the macOS system, probably already worked, I only checked if it regressed.
b) The Python3.8 from the macOS system works after making sure the unusable rpath setting is corrected. Funnily the /usr/bin/python3 has no rpath setting whatsoever, but the library then uses an @rpath pointing to the framework, that is unusable, and the embedding Nuitka compiled program couldn't find the library at all. Strange and messy, but it should be good now.
c) The Python3.8 from CPython (I am assuming that's what `/usr/local/bin/python3` is in the provided VM) needed only the code signature of the python library removed. Since we update the install names, this signature will then prevent it from loading.
d) The Anaconda 3.8 that I downloaded worked after these changes out of the box, probably did before, but now the changes apply to CPython 3.8 too, where they might be necessary in accelerated mode too.

None of these Pythons are what is on the macOS runners from Github actions it seems. Those still work.

I also heavily cleaned it up. To get better bug reports, now every tool called will catch stderr and check it. If an unknown error is uttered, Nuitka will stop on that. Seems the install_name_tool warning that was given plenty about signed code are otherwise easy to overlook.
